src/com/owncloud/android/files/services/TransferRequester.java

```diff
@@ -0,0 +1,317 @@
+/**
+ *  ownCloud Android client application
+ *
+ *  @author David A. Velasco
+ *
+ *  Copyright (C) 2017 ownCloud GmbH.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License version 2,
+ *  as published by the Free Software Foundation.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.owncloud.android.files.services;
+
+import android.accounts.Account;
+import android.app.job.JobInfo;
+import android.app.job.JobScheduler;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Build;
+import android.os.PersistableBundle;
+
+import com.owncloud.android.authentication.AccountUtils;
+import com.owncloud.android.datamodel.OCFile;
+import com.owncloud.android.datamodel.UploadsStorageManager;
+import com.owncloud.android.db.OCUpload;
+import com.owncloud.android.db.UploadResult;
+import com.owncloud.android.lib.common.utils.Log_OC;
+import com.owncloud.android.utils.ConnectivityUtils;
+import com.owncloud.android.utils.PowerUtils;
+import com.owncloud.android.utils.Extras;
+
+import java.net.SocketTimeoutException;
+
+/**
+ * Facade to start operations in transfer services without the verbosity of Android Intents.
+ */
+/**
+ * Facade class providing methods to ease requesting commands to transfer services {@link FileUploader} and
+ * {@link FileDownloader}.
+ *
+ * Protects client objects from the verbosity of {@link android.content.Intent}s.
+ *
+ * TODO add methods for {@link FileDownloader}, right now it's just about uploads
+ */
+
+public class TransferRequester {
+
+    private static final String TAG = TransferRequester.class.getName();
+
+    /**
+     * Call to upload several new files
+     */
+    public void uploadNewFile(
```

Comment:
I would rename this method as uploadNewFiles since it's used for uploading several files.